python/pylibraft/pylibraft/common/cai_wrapper.py

```diff
@@ -71,3 +71,19 @@ def data(self):
         Returns the data pointer of the underlying CUDA array interface
         """
         return self.cai_["data"][0]
+
+    def validate(self, expected_dims=None, expected_dtype=None):
```

Comment:
I like this function but I think the name could be a little more descriptive. Maybe something like validate_shape_dtype?

Reply:
done in last commit